website/docs/migration/1.0.0_to_2.0.0.mdx

```diff
@@ -0,0 +1,246 @@
+---
+title: ^1.0.0 to ^2.0.0
+---
+
+import buildIinit from "./build_init";
+import familyAndDispose from "./family_and_dispose";
+import fromStateProvider from "./from_state_provider";
+import oldLifecycles from "./old_lifecycles";
+import {AutoSnippet} from "../../src/components/CodeSnippet";
+
+Riverpod went `2.0`! To see the full list of changes, consult the [Changelog].
+
+In this page, we will focus on:
+- How to migrate from the deprecated `StateNotifier` to the new `Notifier` APIs
+- How to handle the new `AsyncValue` behavior and its new properties
+- How to migrate from `ProviderListener` to `ProviderObserver` for logging purposes
+
+
+## Migrating from `StateNotifier`
+
+Riverpod `2.0` introduced the new `Notifier` and `AsyncNotifier` classes.  
+These two are meant to replace the legacy `StateNotifier` (and `ChangeNotifier`), with the following advantages:
+- Conciseness and a better DX overall
+- Codegen support
+- One less [external dependency]
+- Easier distinction between sync and async providers
+
+Alas, these features introduce a few breaking changes. This page is meant to help you overcome these.
+
+:::info
+This migration guide can be used for migrating away from `ChangeNotifier` as well.
+Its behavior is quite similar to its equivalent for immutable data, `StateNotifier`.
+:::
+
+### Initialization comes with a `build` method
+
+When defining a `StateNotifierProvider` (or `ChangeNotifierProvider`), initialization is
+typically placed inside the provider itself. Take the following example:
+
+```dart
+final randomProvider = Provider<int>((ref) {
+  return Random().nextInt(6);
+});
+
+class DiceNotifier extends StateNotifier<int> {
+  DiceNotifier(int random) : super(random + 1) {
+    print('Post-initialization effects go here.');
+    _startCamera();
+  }
+
+  void adjust(int offset) => state = state + offset;
+}
+
+final diceNotifierProvider = StateNotifierProvider<DiceNotifier, int>((ref) {
+  final random = ref.watch(randomProvider);
+  return DiceNotifier(random);
+});
+```
+
+With the new `Notifier` class, the above example translates to:
+<AutoSnippet language="dart" {...buildIinit}></AutoSnippet>
+
+You can notice the following differences:
+- A more concise syntax, with no constructor logic
+- There's a new `build` method instead, which specifies initialization logic and its reactive dependencies
+- `NotifierProvider` has no `ref` in its build callback
+- `Notifier` has its own `ref`, internally
+
+### `ref` and family attributes are exposed by their classes directly
```

Comment:
The previous section already talks about how ref is present in notifiers. This new section feels redundant

Reply:
This section also seems to be talking about multiple topics at once

ref and family are distinct subjects. Family should have its own subsection, with crosslinks to the new "passing args to providers" page